### ISSUE TITLE
output the chosen pictures index

### ISSUE THREAD

**Issue by Grenwalls** [chrisgoringe/cg-image-filter#22]
Hey!
Really love the nodes!
I'm trying to use it to select a SEGS from the SEGSdetailer from the impact-pack, but can't find a way really. ( I am aware of the segs picker node from the impact-pack, but having to run the entire workflow twice to pick which output is wanted is very annyoing and time consuming)

It would be very useful if the Image Filter node could have an output with just the integer of the chosen pictures index in the batch, allowing the Image Filter node to pick which segspicker to choose.
That would open up for a whole lot of fun things! :)

I've tried to use the MASK-input/output and converting to segs etc but it's just not really working.

**Comment by Grenwalls** [comment]
I coded it into the node myself, turned out to be more simple than I thought!


If anyone else wants to do the same, I changed the class ImageFilter(PreviewImage) in image_filter.py to 

```

class ImageFilter(PreviewImage):
    RETURN_TYPES = ("IMAGE", "LATENT", "MASK", "STRING", "STRING")
    RETURN_NAMES = ("images", "latents", "masks", "indices_original", "indices_shifted")
    FUNCTION = "func"
    CATEGORY = "image_filter"
    OUTPUT_NODE = False

    @classmethod
    def INPUT_TYPES(s):
        return {
            "required": { 
                "images": ("IMAGE", ), 
                "timeout": ("INT", {"default": 60, "tooltip": "Timeout in seconds."}),
                "ontimeout": (["send none", "send all", "send first", "send last"], {}),
            },
            "optional": {
                "latents": ("LATENT", {"tooltip": "Optional - if provided, will be output"}),
                "masks": ("MASK", {"tooltip": "Optional - if provided, will be output"}),
            },
            "hidden": HIDDEN,
        }
    
    @classmethod
    def IS_CHANGED(cls, **kwargs):
        return float("NaN")
    
    def func(self, images, timeout, ontimeout, uid, latents=None, masks=None, **kwargs):
        B = images.shape[0]
        all_the_same = (B and all((images[i] == images[0]).all() for i in range(1, B))) 
        urls: list[str] = self.save_images(images=images, **kwargs)['ui']['images']
        payload = {"uid": uid, "urls": urls, "allsame": all_the_same}

        response = send_with_resend(payload, timeout, uid)

        if not response:
            if ontimeout == 'send none': response = ""
            if ontimeout == 'send all': response = ",".join(str(x) for x in range(len(images)))
            if ontimeout == 'send first': response = "0"
            if ontimeout == 'send last': response = str(len(images) - 1)

        images_to_return = [int(x) for x in response.split(",") if x]

        if len(images_to_return) == 0: 
            raise InterruptProcessingException()

        # Generate both unshifted and shifted index strings
        indices_original = ",".join(str(i) for i in images_to_return)
        indices_shifted = ",".join(str(i + 1) for i in images_to_return)

        images = torch.stack([images[i] for i in images_to_return])
        latents = {"samples": torch.stack([latents['samples'][i] for i in images_to_return])} if latents is not None else None
        masks = torch.stack([masks[i] for i in images_to_return]) if masks is not None else None

        return images, latents, masks, indices_original, indices_shifted

```

This returns the selected indexes as a string separated by a comma. Also a separate +1 shifted output since the segspicker node wanted it that way.
